apps/io-profile-async/src/functions/__tests__/expired-sessions-discoverer.test.ts

```diff
@@ -0,0 +1,550 @@
+        },
+        tagOverrides: {
+          samplingEnabled: "false"
+        }
+      });
+      expect(E.isRight(result)).toBe(true);
+    });
+  });
+
+  describe("processChunk", () => {
+    it("should succeed when processItem succeed on each item", async () => {
+      const chunk = [item, item, item];
+      const result = await processChunk(chunk)(baseDeps)();
+
+      expect(
+        mockSessionNotificationsRepository.updateExpiredSessionNotificationFlag
+      ).toHaveBeenCalledTimes(chunk.length);
+      expect(
+        mockExpiredUserSessionsQueueRepository.sendExpiredUserSession
+      ).toHaveBeenCalledTimes(chunk.length);
+
+      expect(result).toStrictEqual(E.right(void 0));
+    });
+
+    it("should fail when processItem fails on an item", async () => {
+      const aCosmosError = { kind: "COSMOS_ERROR_RESPONSE" } as CosmosErrors;
+
+      updateExpiredSessionNotificationFlagMock.mockImplementationOnce(() =>
+        TE.left(aCosmosError)
+      );
+
+      const chunk = [item, item, item];
+      const result = await processChunk(chunk)(baseDeps)();
+
+      expect(
+        mockSessionNotificationsRepository.updateExpiredSessionNotificationFlag
+      ).toHaveBeenCalledTimes(chunk.length);
+      expect(
+        mockExpiredUserSessionsQueueRepository.sendExpiredUserSession
+      ).toHaveBeenCalledTimes(chunk.length - 1); // -1 for the failed item
+      expect(trackEventMock).not.toHaveBeenCalled();
+
+      expect(result).toStrictEqual(
+        E.left([
+          new TransientError(
+            `Error updating expired session flag(EXPIRED_SESSION): ${aCosmosError.kind}`
+          )
+        ])
+      );
+    });
+
+    it("should fail when processItem fails on some items, reporting the errors", async () => {
+      const aCosmosError = { kind: "COSMOS_ERROR_RESPONSE" } as CosmosErrors;
+
+      updateExpiredSessionNotificationFlagMock
+        .mockImplementationOnce(() => TE.left(aCosmosError))
+        .mockImplementationOnce(() => TE.left(aCosmosError));
+
+      const chunk = [item, item, item];
+      const result = await processChunk(chunk)(baseDeps)();
+
+      expect(
+        mockSessionNotificationsRepository.updateExpiredSessionNotificationFlag
+      ).toHaveBeenCalledTimes(chunk.length);
+      expect(
+        mockExpiredUserSessionsQueueRepository.sendExpiredUserSession
+      ).toHaveBeenCalledTimes(chunk.length - 2); // -2 for the failed items
+      expect(trackEventMock).not.toHaveBeenCalled();
+
+      expect(result).toStrictEqual(
+        E.left([
+          new TransientError(
+            `Error updating expired session flag(EXPIRED_SESSION): ${aCosmosError.kind}`
+          ),
+          new TransientError(
+            `Error updating expired session flag(EXPIRED_SESSION): ${aCosmosError.kind}`
+          )
+        ])
+      );
+    });
+
+    it("should succeed when a flag revert fails", async () => {
```

Comment:
This scenario should be covered in the `processItem` test suite.

In this suite, it can be treated as a successful result from `processItem` itself.
This highlights the value of mocking `processItem`.